### PR TITLE
Backend Script Fixes

### PR DESCRIPTION
**Pull request by shravan1k**
### Administrative Info
Monday Board ID: 853648344
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Check for differences between database and api response before attempting to update jobs.
- Script now sends emails for classes that start in the range of 15 mins to 30 mins after present time.
- No longer inserts jobs from api that are happening within 15 mins from present time.
- Updates statuses of jobs to success or fail after attempt to send reminder emails.

### Testing
How did you confirm your changes worked? 
- To confirm that updates were occurring properly, I modified values in the database and then ran the script. Only the jobs that were changed were being passed into updateJobs and the updates were reflected in the database.
- To confirm that the emails were being sent only once and that the statuses of the jobs were updating properly, I added a client to the database for a job that was happening between 15 and 30 mins. The email only sent once and the database correctly updated.

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

If just an id is printed, it means that the job with the id was passed into updateJobs.

| When job needs to be updated | When job is already up to date |
| --- | --- |
| Screen Shot 2021-05-09 at 11 51 07 AM | Screen Shot 2021-05-09 at 11 50 11 AM |

### CODE REVIEW

**Review by eliasfang — APPROVED** [review]
Nice!

File: app/models/jobsModel.js [hunk 1]
```diff
@@ -135,31 +127,58 @@ async function getAll(){
...
  * Returns a list of jobs of the classes that need to be
- * dequeued within a specfic time frame
+ * dequeued within a specfic time frame.
  *
- * @param time integer specifying time frame
+ * @param minMinutes lower bound of minutes after present time that is included in range
+ * @param maxMinutes upper bound of minutes after present time that is included in range
  */
-async function getByMinutesFromNow(mins){
+async function getByMinutesFromNow(minMinutes, maxMinutes){
```
Comment (mingkaili) [inline app/models/jobsModel.js]: rename to getByMinutesInRange

File: app/models/jobsModel.js [hunk 2]
```diff
@@ -135,31 +127,58 @@ async function getAll(){
...
 }
 
+/**
+ * Updates the job's status depending on if reminder emails
+ * were successfully sent or not.
+ * 
+ * @param {*} jobId Id of the job to update.
+ * @param {*} isSuccess boolean of whether emails were successfully sent.
+ */
+async function updateJobStatus(jobId, isSuccess){
```
Comment (mingkaili) [inline app/models/jobsModel.js]: Usually, we don't pass the condition in as a parameter, the following format is better: 

```
if(isSuccess){
  await jobModel.updateById();
}
```

Also, name this function updateById, you don't need a separate function for updating one single field

File: modules/mailer.js [hunk 3]
```diff
@@ -36,9 +37,11 @@ async function sendReminders(classInfo, clientEmails){
                 html: data
             }, async function(err, data){
                 if (err) {
+                    jobsModel.updateJobStatus(classInfo.id, false);
```
Comment (mingkaili) [inline modules/mailer.js]: name change

File: modules/mailer.js [hunk 4]
```diff
@@ -36,9 +37,11 @@ async function sendReminders(classInfo, clientEmails){
...
                     console.error(err, "Email failed");
                 }
                 else {
+                    jobsModel.updateJobStatus(classInfo.id, true);
```
Comment (mingkaili) [inline modules/mailer.js]: name change

**Review by mingkaili — CHANGES_REQUESTED** [review]

**Review by mingkaili — APPROVED** [review]

**Pull request merged**